yt/data_objects/data_containers.py

```diff
@@ -703,13 +704,14 @@ def to_glue(self, fields, label="yt", data_collection=None):
 
     def create_firefly_object(
         self,
-        JSONdir,
+        datadir=None,
         fields_to_include=None,
         fields_units=None,
         default_decimation_factor=100,
         velocity_units="km/s",
         coordinate_units="kpc",
         show_unused_fields=0,
+        JSONdir=None,
```

Comment:
we don't want this deprecated parameter to be valid if passed as _positional_, so let's mark it as keyword only
```suggestion
        *,
        JSONdir=None,
```